service/cast.go

```diff
@@ -0,0 +1,49 @@
+package service
+
+import (
+	"fmt"
+	"strconv"
+)
+
+// Cast converts map[string]string to map[string]interface{} based on defined types in the service tasks map.
+func (s *Service) Cast(taskKey string, taskData map[string]string) (map[string]interface{}, error) {
+	task, ok := s.Tasks[taskKey]
+	if !ok {
+		return nil, fmt.Errorf("Task %q not found", taskKey)
+	}
+
+	m := make(map[string]interface{}, len(taskData))
+	for key, value := range taskData {
+		inputType, ok := task.Inputs[key]
+		if !ok {
+			return nil, fmt.Errorf("Task %q has no input named %q", taskKey, key)
+		}
+
+		switch inputType.Type {
+		case "String":
+			m[key] = value
+		case "Number":
+			i, err := strconv.ParseInt(value, 10, 64)
+			if err != nil {
+				f, err1 := strconv.ParseFloat(value, 64)
+				if err1 != nil {
+					return nil, fmt.Errorf("Task %q - input %q is not a Number type", taskKey, key)
+				}
+				m[key] = f
+			} else {
+				m[key] = i
+			}
+		case "Boolean":
+			b, err := strconv.ParseBool(value)
+			if err != nil {
+				return nil, fmt.Errorf("Task %q - input %q is not a Boolean type", taskKey, key)
+			}
+			m[key] = b
+		case "Object":
+			return nil, fmt.Errorf("Task %q - input %q - Object type is not supported", taskKey, key)
```

Comment:
Can we really not pass any Object with the CLI? Even a JSON stringified?

Reply:
Ok, so I'll continue here instead of return an error